modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase

```diff
@@ -0,0 +1,211 @@
+@component-name = "portal-headless"
+definition {
+
+	property custom.properties = "feature.flag.LPS-184413=true${line.separator}feature.flag.LPS-167253=true${line.separator}feature.flag.LPS-178642=true${line.separator}feature.flag.LPS-186757=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless Builder";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginUI();
+
+		task ("Given create an API application") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "test",
+				status = "unpublished",
+				title = "test");
+
+			ApplicationAPI.setUpGlobalAPIApplicationId();
+		}
+	}
+
+	tearDown {
+		ApplicationAPI.deleteAllAPIApplication();
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 4
+	test CanDeleteAPIFromApiBuilderAndApi {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("And Given from dropdown menu I click Edit option") {
+			APIBuilderUI.goToDeleteAPIApplication(key_title = "test");
+		}
+
+		task ("When I confirm deletion by typing the API title") {
+			Type(
+				locator1 = "APIBuilder#PLACEHOLDER_DELETE_APPLICATION",
+				value1 = "test");
+
+			Click(locator1 = "APIBuilder#MODAL_DELETE_BUTTON");
+		}
+
+		task ("Then the API is not present in API Builder") {
+			AssertElementNotPresent(
+				key_name = "test",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
+		}
+
+		task ("And Then the API is removed from API") {
+			var response = ApplicationAPI.getAPIApplications();
+
+			var totalCount = JSONUtil.getWithJSONPath(${response}, "$..totalCount");
+
+			TestUtils.assertEquals(
+				actual = ${totalCount},
+				expected = 0);
+		}
+	}
+
+	@priority = 4
+	test CanDeletePublishedApiAndAssociatedEntities {
```

Comment:
Just wondering maybe we can move CanDeletePublishedApiAndAssociatedEntities and the following CanDeleteUnpublishedApiAndAssociatedEntities into interation test? @magjed4289

Reply:
yes if it is only API layer test we can move it to the integration: analysing first whether a similar test is already there.

Reply:
but this story is about the UI so I don't understand why the deletion is being made through API

Reply:
good point, yes, we should update to delete by UI, thanks!